CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -103,9 +103,12 @@ public async Task<Result<LessonDto>> CreateLessonAsync(CreateLessonDto lessonDto
             }
         }
 
-        public async Task<Result<IList<LessonDto>>> GetAllLessonsAsync()
+        public async Task<Result<IList<LessonDto>>> GetLessonsByDate(DateTime? startDate, DateTime? finishDate)
         {
-            var lessons = await _unitOfWork.LessonRepository.GetAllAsync();
+            var lastLesson = await _unitOfWork.LessonRepository.GetLastLesson();
```

Comment:
If startDate and finishDate are null then compute them as you do in the repository.